rfcs/text/0000-matrix-service.md

```diff
@@ -0,0 +1,418 @@
+    "field_name": "genes_detected",
+    "field_type": "numeric",
+    "maximum": 13108,
+    "minimum": 2
+}
+```
+
+### Cell-centric Index
+
+To realize the metadata querying interface required by the API, the Matrix Service maintains an index of cells capable of answering questions about a cell's experimental and scientific background. In particular, a cell is represented as the fundamental entity in the database and maintains foreign key references to various tables representing metadata facets. To promote readability of the index, table field names are simplified versions of their full canonical metadata names. Below is the current database table structure:
+
+*Table 2.1 Cell-centric index schema*
+
+| Table name          | Fields                                                       | Canonical Metadata Name                                      | Description                                                  |
+| ------------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ |
+| cell                | cellkey (PK)<br />cellsuspensionkey (FK)<br/>projectkey (FK)<br/>librarykey (FK)<br/>analysiskey (FK)<br/>file_uuid<br/>file_version<br/>barcode<br/>genes_detected<br/>total_umis<br/>emptydrops_is_cell | N/A (derived field)<br />cell_suspension.provenance.document_id<br/>project.provenance.document_id<br/>library_preparation_protocol.provenance.document_id<br />analysis_protocol.provenance.document_id<br />file_uuid<br />file_version<br />N/A (not metadata)<br />N/A (not metadata)<br />N/A (not metadata)<br />N/A (not metadata) | Each row represents a single cell and is described by foreign key references to other facet tables representing metadata fields. |
+| expression          | cellkey (PK)<br />featurekey (FK)<br/>exprtype<br/>exprvalue      | N/A (derived field)<br />N/A (not metadata)<br />N/A (not metadata)<br />N/A (not metadata) | Stores gene expression values. Each row specifies a cell, feature (e.g. gene) and expression value. |
+| feature             | featurekey (PK)<br />featurename<br/>featuretype<br/>chromosome<br/>featurestart<br/>featureend<br/>isgene<br/>genus_species | N/A (not metadata)<br />N/A (not metadata)<br />N/A (not metadata)<br />N/A (not metadata)<br />N/A (not metadata)<br />N/A (not metadata)<br />N/A (not metadata)<br />cell_suspension.genus_species.ontology_label | Describes gene and transcript information for supported species. Data is populated from GENCODE GTF annotation files. |
+| analysis            | analysiskey (PK)<br />bundle_fqid<br/>bundle_uuid<br/>bundle_version<br/>protocol<br/>awg_disposition | analysis_protocol.provenance.document_id<br />dss_bundle_fqid<br />bundle_uuid<br />bundle_version<br />analysis_protocol.protocol_core.protocol_id<br />analysis_working_group_approval_status | Represents an analysis bundle.                               |
```

Comment:
We do not provide a field for `analysis_working_group_approval_status` because there is no way to update that value when the status of a pipeline changes, so we are only running approved pipelines for now.

Reply:
:+1: we basically always [set this field](https://github.com/HumanCellAtlas/matrix-service/blob/master/matrix/common/etl/transformers/analysis.py#L32) to `blessed` for the time being.

Reply:
perfect!

Reply:
Yup, it is.

Reply:
[Ticket](https://app.zenhub.com/workspaces/dcp-5ac7bcf9465cb172b77760d9/issues/humancellatlas/matrix-service/429). Resolving